### PR TITLE
Add metrics service port name as Helm chart parameter (#165)

### PR DESCRIPTION
**Pull request by maurizio-lattuada**
Add parameter `portName` for the metrics service in all four charts.

I executed a deployment on my local k8s cluster (created with Docker for Mac) only for the Hazelcast chart having this values file called `values.yaml`, wherein I set the new parameter `portName` equals to `metrics8088`:
```yaml
metrics:
  enabled: true
  service:
    type: ClusterIP
    port: 8088
    portName: metrics8088
    annotations:
      prometheus.io/scrape: "true"
      prometheus.io/path: "/metrics"
      prometheus.io/port: "8088"
```

Here the outcome of `helm install --dry-run`, please look for `metrics8088`:
```shell script
MacBook-Pro-di-Maurizio:helm-chart maurizio$ helm install hazelcast stable/hazelcast -f values.yaml --dry-run -n hzc
NAME: hazelcast
LAST DEPLOYED: Mon Oct 26 10:39:18 2020
NAMESPACE: hzc
STATUS: pending-install
REVISION: 1
HOOKS:
---
# Source: hazelcast/templates/tests/test-hazelcast.yaml
apiVersion: v1
kind: Pod
metadata:
  name: "hazelcast-test-3m2oq"
  annotations:
    "helm.sh/hook": test-success
    "helm.sh/hook-delete-policy": hook-succeeded, hook-failed
  labels:
    app.kubernetes.io/name: hazelcast
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/component: "test"
    role: test
spec:
  hostNetwork: false
  hostPID: false
  hostIPC: false
  securityContext:
    runAsNonRoot: true
    runAsUser: 1001
    runAsGroup: 1001
  containers:
  - name: "hazelcast-test"
    image: "hazelcast/hazelcast:4.0.3"
    command:
    - "bash"
    - "-c"
    - |
      set -ex
      # Get the number of Hazelcast members in the cluster
      CLUSTER_SIZE=$(curl hazelcast:5701/hazelcast/health/cluster-size)
      # Test the currect number of Hazelcast members
      test ${CLUSTER_SIZE} -eq 3
    securityContext:
      runAsNonRoot: true
      runAsUser: 1001
      runAsGroup: 1001
      privileged: false
      readOnlyRootFilesystem: true
      allowPrivilegeEscalation: false
      capabilities:
        drop:
        - ALL
  restartPolicy: Never
---
# Source: hazelcast/templates/tests/test-management-center.yaml
apiVersion: v1
kind: Pod
metadata:
  name: "hazelcast-mancenter-test-7rfvg"
  annotations:
    "helm.sh/hook": test-success
    "helm.sh/hook-delete-policy": hook-succeeded, hook-failed
  labels:
    app.kubernetes.io/name: hazelcast
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/component: "test"
    role: test
spec:
  hostNetwork: false
  hostPID: false
  hostIPC: false
  securityContext:
    runAsNonRoot: true
    runAsUser: 1001
    runAsGroup: 1001
  containers:
  - name: "hazelcast-mancenter-test"
    image: "hazelcast/hazelcast:4.0.3"
    command:
    - "bash"
    - "-c"
    - |
      set -ex
      # Get the HTTP Response Code of the Health Check
      HEALTH_CHECK_HTTP_RESPONSE_CODE=$(curl --write-out %{http_code} --silent --output /dev/null hazelcast-mancenter:8080/health)
      # Test the currect number of Hazelcast members
      test ${HEALTH_CHECK_HTTP_RESPONSE_CODE} -eq 200
    securityContext:
      runAsNonRoot: true
      runAsUser: 1001
      runAsGroup: 1001
      privileged: false
      readOnlyRootFilesystem: true
      allowPrivilegeEscalation: false
      capabilities:
        drop:
        - ALL
  restartPolicy: Never
MANIFEST:
---
# Source: hazelcast/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: hazelcast
  labels:
    app.kubernetes.io/name: hazelcast
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
---
# Source: hazelcast/templates/config.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: hazelcast-configuration
  labels:
    app.kubernetes.io/name: hazelcast
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
data:
  hazelcast.yaml: |-
    hazelcast:
      network:
        join:
          kubernetes:
            enabled: true
            namespace: ${namespace}
            service-name: ${serviceName}
          multicast:
            enabled: false
        rest-api:
          enabled: true
          endpoint-groups:
            HEALTH_CHECK:
              enabled: true
---
# Source: hazelcast/templates/mancenter-config.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: hazelcast-mancenter-configuration
  labels:
    app.kubernetes.io/name: hazelcast-mancenter
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
data:
  hazelcast-client.yaml: |-
    hazelcast-client:
      network:
        kubernetes:
          enabled: true
          namespace: ${namespace}
          service-name: ${serviceName}
---
# Source: hazelcast/templates/role.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: hazelcast-hzc
  labels:
    app.kubernetes.io/name: hazelcast
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
rules:
- apiGroups:
  - ""
  resources:
  - endpoints
  - pods
  - nodes
  verbs:
  - get
  - list
---
# Source: hazelcast/templates/rolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: hazelcast-hzc
  labels:
    app.kubernetes.io/name: hazelcast
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: hazelcast-hzc
subjects:
- kind: ServiceAccount
  name: hazelcast
  namespace: hzc
---
# Source: hazelcast/templates/mancenter-service.yaml
apiVersion: v1
kind: Service
metadata:
  name: hazelcast-mancenter
  labels:
    app.kubernetes.io/name: hazelcast-mancenter
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
spec:
  type: LoadBalancer
  selector:
    app.kubernetes.io/name: hazelcast-mancenter
    app.kubernetes.io/instance: "hazelcast"
    role: mancenter
  ports:
  - protocol: TCP
    port: 8080
    targetPort: mancenter
    name: http
  - protocol: TCP
    port: 443
    targetPort: mancenter
    name: https
---
# Source: hazelcast/templates/metrics-service.yaml
apiVersion: v1
kind: Service
metadata:
  name: hazelcast-metrics
  labels:
    app.kubernetes.io/name: hazelcast
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
  annotations:
    prometheus.io/path: /metrics
    prometheus.io/port: "8088"
    prometheus.io/scrape: "true"
spec:
  type: ClusterIP
  selector:
    app.kubernetes.io/name: hazelcast
    app.kubernetes.io/instance: "hazelcast"
  ports:
  - protocol: TCP
    port: 8088
    targetPort: metrics8088
    name: metrics8088
---
# Source: hazelcast/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: hazelcast
  labels:
    app.kubernetes.io/name: hazelcast
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
spec:
  type: ClusterIP
  clusterIP: None
  selector:
    app.kubernetes.io/name: hazelcast
    app.kubernetes.io/instance: "hazelcast"
    role: hazelcast
  ports:
  - protocol: TCP
    port: 5701
    targetPort: hazelcast
    name: hzport
---
# Source: hazelcast/templates/mancenter-statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: hazelcast-mancenter
  labels:
    app.kubernetes.io/name: hazelcast-mancenter
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
spec:
  serviceName: hazelcast-mancenter
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: hazelcast-mancenter
      app.kubernetes.io/instance: "hazelcast"
      role: mancenter
  template:
    metadata:
      labels:
        app.kubernetes.io/name: hazelcast-mancenter
        helm.sh/chart: hazelcast-3.4.12
        app.kubernetes.io/instance: "hazelcast"
        app.kubernetes.io/managed-by: "Helm"
        role: mancenter
    spec:
      hostNetwork: false
      hostPID: false
      hostIPC: false
      securityContext:
        runAsNonRoot: true
        runAsUser: 65534
        runAsGroup: 65534
        fsGroup: 65534
      containers:
      - name: hazelcast-mancenter
        image: "hazelcast/management-center:4.2020.08"
        imagePullPolicy: "IfNotPresent"
        resources:
          null
        ports:
        - name: mancenter
          containerPort: 8080
        livenessProbe:
          httpGet:
            path: /health
            port: 8081
          initialDelaySeconds: 30
          periodSeconds: 10
          timeoutSeconds: 5
          successThreshold: 1
          failureThreshold: 3
        readinessProbe:
          httpGet:
            path: /health
            port: 8081
          initialDelaySeconds: 30
          periodSeconds: 10
          timeoutSeconds: 1
          successThreshold: 1
          failureThreshold: 3
        volumeMounts:
        - name: config
          mountPath: /config
        - name: mancenter-storage
          mountPath: /data
        env:
        - name: MC_LICENSE_KEY
          value:
        - name: MC_INIT_CMD
          value: "./mc-conf.sh cluster add --lenient=true -H /data -cc /config/hazelcast-client.yaml"
        - name: JAVA_OPTS
          value: " -Dhazelcast.mc.healthCheck.enable=true -DserviceName=hazelcast -Dnamespace=hzc -Dhazelcast.mc.tls.enabled=false "
        securityContext:
          runAsNonRoot: true
          runAsUser: 65534
          runAsGroup: 65534
          privileged: false
          readOnlyRootFilesystem: false
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
      securityContext:
        fsGroup: 65534
      serviceAccountName: hazelcast
      volumes:
      - name: config
        configMap:
           name: hazelcast-mancenter-configuration
      - name: mancenter-storage

  volumeClaimTemplates:
  - metadata:
      name: mancenter-storage
      labels:
        app.kubernetes.io/name: hazelcast-mancenter
        helm.sh/chart: "hazelcast-3.4.12"
        app.kubernetes.io/instance: "hazelcast"
        app.kubernetes.io/managed-by: "Helm"
    spec:
      accessModes:
        - "ReadWriteOnce"
      resources:
        requests:
          storage: "8Gi"
---
# Source: hazelcast/templates/statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: hazelcast
  labels:
    app.kubernetes.io/name: hazelcast
    helm.sh/chart: hazelcast-3.4.12
    app.kubernetes.io/instance: "hazelcast"
    app.kubernetes.io/managed-by: "Helm"
spec:
  serviceName: hazelcast
  replicas: 3
  selector:
    matchLabels:
      app.kubernetes.io/name: hazelcast
      app.kubernetes.io/instance: "hazelcast"
      role: hazelcast
  template:
    metadata:
      labels:
        app.kubernetes.io/name: hazelcast
        helm.sh/chart: hazelcast-3.4.12
        app.kubernetes.io/instance: "hazelcast"
        app.kubernetes.io/managed-by: "Helm"
        role: hazelcast
      annotations:
        checksum/hazelcast-config: b6d39a9c43a56af9ebb59e1a70e6467460996d848dabf9789e06e7e817f5ff37
    spec:
      terminationGracePeriodSeconds: 600
      hostNetwork: false
      hostPID: false
      hostIPC: false
      securityContext:
        runAsNonRoot: true
        runAsUser: 65534
        runAsGroup: 65534
        fsGroup: 65534
      containers:
      - name: hazelcast
        image: "hazelcast/hazelcast:4.0.3"
        imagePullPolicy: "IfNotPresent"
        resources:
          null
        ports:
        - name: hazelcast
          containerPort: 5701
          hostPort:
        - name: metrics8088
          containerPort: 8088
        livenessProbe:
          httpGet:
            path: /hazelcast/health/node-state
            port: 5701
          initialDelaySeconds: 30
          periodSeconds: 10
          timeoutSeconds: 10
          successThreshold: 1
          failureThreshold: 10
        readinessProbe:
          httpGet:
            path: /hazelcast/health/ready
            port: 5701
          initialDelaySeconds: 30
          periodSeconds: 10
          timeoutSeconds: 10
          successThreshold: 1
          failureThreshold: 10
        volumeMounts:
        - name: hazelcast-storage
          mountPath: /data/hazelcast
        env:
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: PROMETHEUS_PORT
          value: "8088"
        - name: JAVA_OPTS
          value: "-Dhazelcast.config=/data/hazelcast/hazelcast.yaml -DserviceName=hazelcast -Dnamespace=hzc -Dhazelcast.shutdownhook.policy=GRACEFUL -Dhazelcast.shutdownhook.enabled=true -Dhazelcast.graceful.shutdown.max.wait=600  -Dhazelcast.jmx=true "
        securityContext:
          runAsNonRoot: true
          runAsUser: 65534
          runAsGroup: 65534
          privileged: false
          readOnlyRootFilesystem: true
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
      serviceAccountName: hazelcast
      volumes:
      - name: hazelcast-storage
        configMap:
          name: hazelcast-configuration

NOTES:
** Hazelcast cluster is being deployed! **

-------------------------------------------------------------------------------

To access Hazelcast within the Kubernetes cluster:

- Use Hazelcast Client with Kubernetes Discovery Strategy. Read more at: https://github.com/hazelcast/hazelcast-kubernetes.

-------------------------------------------------------------------------------

To access Hazelcast from outside the Kubernetes cluster:

- Use Hazelcast Client with Smart Routing disabled:
  *) Forward port from POD:
     $ export POD=$(kubectl get pods --namespace hzc -l "app.kubernetes.io/name=hazelcast,role=hazelcast" -o jsonpath="{.items[0].metadata.name}")
     $ kubectl port-forward --namespace hzc $POD 5701:5701
  *) In Hazelcast Client configure:
     clientConfig.getNetworkConfig().setSmartRouting(false);
     clientConfig.getNetworkConfig().addAddress("127.0.0.1:5701");

-------------------------------------------------------------------------------

To access Hazelcast Management Center:
  *) Check Management Center external IP:
     $ export MANCENTER_IP=$(kubectl get svc --namespace hzc hazelcast-mancenter -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
  *) Open Browser at: http://$MANCENTER_IP:8080
```

Here the outcome, once deployed without `--dry-run`, of `kubectl describe service hazelcast-metrics`; here again, the `Port` and `TargetPort` have now name `metrics8088`, as it has been declared in `values.yaml` above:
```shell script
MacBook-Pro-di-Maurizio:helm-chart maurizio$ kubectl describe service hazelcast-metrics
Name:              hazelcast-metrics
Namespace:         hzc
Labels:            app.kubernetes.io/instance=hazelcast
                   app.kubernetes.io/managed-by=Helm
                   app.kubernetes.io/name=hazelcast
                   helm.sh/chart=hazelcast-3.4.12
Annotations:       meta.helm.sh/release-name: hazelcast
                   meta.helm.sh/release-namespace: hzc
                   prometheus.io/path: /metrics
                   prometheus.io/port: 8088
                   prometheus.io/scrape: true
Selector:          app.kubernetes.io/instance=hazelcast,app.kubernetes.io/name=hazelcast
Type:              ClusterIP
IP:                10.96.51.226
Port:              metrics8088  8088/TCP
TargetPort:        metrics8088/TCP
Endpoints:         10.1.0.10:8088,10.1.0.12:8088,10.1.0.13:8088
Session Affinity:  None
Events:            <none>
```

Finally, by doing a port forward, we have effectively all the Hazelcast metrics:
localhost_8088_metrics

Hope it is enough and right 👍

### CODE REVIEW

**Comment by hasancelik** [comment]
verify

File: stable/hazelcast-enterprise/values.yaml [hunk 1]
```diff
@@ -219,6 +219,7 @@ metrics:
   service:
     type: ClusterIP
     port: 8080
+    portName: metrics8080
```
Comment (hasancelik) [inline stable/hazelcast-enterprise/values.yaml:222]: `portName` is configurable so I think we can use still `metrics` as a default port name.

**Review by hasancelik — APPROVED** [review]
Thanks a lot for your contribution @maurizio-lattuada 💯 

LGTM 👍 with just one minor comment, PTAL

File: stable/hazelcast-enterprise/values.yaml (hunk 1 above)
Comment (maurizio-lattuada) [reply stable/hazelcast-enterprise/values.yaml:222]: well yes, it makes definitely sense to leave it equals to `metrics` by default :+1: 
Gonna change it right now.

**Review by hasancelik — APPROVED** [review]

**Pull request merged**